src/modules/unleash.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+let unleashClient = null;
+
+export function setUnleashClient(_unleashClient) {
+	unleashClient = _unleashClient;
+}
+
+export function getUnleashClient() {
+	return unleashClient;
+}
```

Comment:
I know we did this in the wallet singleton, but I think we should refactor this singleton (and the wallet singleton) so we don't pollute the global context... Something like:

```javascript
const unleashModule = (function() {
    let unleashClient = null;

    function setUnleashClient(_unleashClient) {
        unleashClient = _unleashClient;
    }

    function getUnleashClient() {
        return unleashClient;
    }

    return {
        setUnleashClient,
        getUnleashClient
    };
})();

export default unleashModule;
```

What do you think?

Reply:
As we've talked, this application already uses ES6 modules through NodeJs 20. The variables and properties do not pollute the global environment when they're declared as they are.

See [Mdn Web Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#other_differences_between_modules_and_standard_scripts):
> ...module features are imported into the scope of a single script — they aren't available in the global scope. Therefore, you will only be able to access imported features in the script they are imported into...

Considering this, I believe it's better to keep the code as it is for legibility.
Do you agree?